workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourManager.cs

```diff
@@ -1,60 +1,202 @@
 using System.Collections.Generic;
+using Improbable.Gdk.Core.MonoBehaviours;
 using Improbable.Worker;
 using UnityEngine;
+using Entity = Unity.Entities.Entity;
 
 namespace Improbable.Gdk.Core
 {
-    public class ReaderWriterImpl
-    {
-        // This is placeholder code.
-    }
-
     /// <summary>
-    ///     Invokes dispatcher callbacks on GameObjects that represent entities.
+    ///     Keeps track of Reader/Writer availability for SpatialOSBehaviours on a particular GameObject and decides when
+    ///     a SpatialOSBehaviour should be enabled, calling into the SpatialOSBehaviourLibrary for injection.
     /// </summary>
-    public class SpatialOSBehaviourManager
+    internal class SpatialOSBehaviourManager
     {
-        public GameObject GameObject;
-        private Dictionary<uint, List<MonoBehaviour>> componentIdToReaderSpatialOSBehaviours;
-        private Dictionary<uint, List<MonoBehaviour>> componentIdToWriterSpatialOSBehaviours;
-        private Dictionary<MonoBehaviour, int> numUnsatisfiedComponents;
-        private List<MonoBehaviour> spatialOSBehavioursToEnable;
-        private List<MonoBehaviour> spatialOSBehavioursToDisable;
-        private Dictionary<uint, Dictionary<MonoBehaviour, ReaderWriterImpl>> readersWriters;
+        private readonly Entity entity;
+        private readonly long spatialId;
+
+        private readonly Dictionary<uint, HashSet<MonoBehaviour>> componentIdToReaderSpatialOSBehaviours
```

Comment:
These variables are quite verbose - my eyes glaze over every time I read them. Is there some abbreviation that could be applied here and below?
(Also, should be `behavior`)

Reply:
I'll change them, in some places it's definitely a tradeoff of verbose and expressive vs short and unclear, though. Check back to see how you like whatever I come up with

Reply:
I think behaviour is the one English spelling that we use, because MonoBehaviour uses it :P